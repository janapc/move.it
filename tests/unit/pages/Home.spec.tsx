import { render } from "@testing-library/react";

import Home from "../../../src/pages/index";

global.Notification = ({
  requestPermission: jest.fn(),
  permission: "granted"
} as unknown) as jest.Mocked<typeof Notification>;

describe("Home", () => {
  it("Should render page Home correctly", () => {
    const wrapper = render(
      <Home level={2} currentExperience={30} challengesCompleted={3} />
    );

    expect(wrapper.asFragment()).toMatchSnapshot()
  });
});
