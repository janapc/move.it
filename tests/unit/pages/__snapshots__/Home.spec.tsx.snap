// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Should render page Home correctly 1`] = `
<DocumentFragment>
  <div>
    <header>
      <span>
        0 xp
      </span>
      <div>
        <div
          data-testid="bar-width-percent"
          style="width: 20.833333333333332%;"
        />
        <span
          data-testid="bar-left-percent"
          style="left: 20.833333333333332%;"
        >
          30 xp
        </span>
      </div>
      <span
        data-testid="expirence-to-next-level"
      >
        144 xp
      </span>
    </header>
    <section>
      <div>
        <div>
          <img
            alt="Janaina Pedrina"
            src="https://github.com/janapc.png"
          />
          <div
            data-testid="profile"
          >
            <strong
              data-testid="name"
            >
              Janaina Pedrina
            </strong>
            <p
              data-testid="level"
            >
              <img
                alt="level"
                src="icons/level.svg"
              />
              Level 2
            </p>
          </div>
        </div>
        <div>
          <span>
            Desafios completos
          </span>
          <span
            data-testid="challengesCompleted"
          >
            3
          </span>
        </div>
        <div>
          <div
            data-testid="time"
          >
            <div>
              <span>
                0
              </span>
              <span>
                0
              </span>
            </div>
            <span>
              :
            </span>
            <div>
              <span>
                0
              </span>
              <span>
                3
              </span>
            </div>
          </div>
          <button
            data-testid="btn-initial"
            type="button"
          >
            Iniciar um ciclo 
            <img
              src="icons/play.svg"
            />
          </button>
        </div>
      </div>
      <div>
        <div>
          <div>
            <strong>
              Finalize um ciclo para recenber um desafio
            </strong>
            <p>
              <img
                alt="Level Up"
                src="icons/level-up.svg"
              />
              Avance de level completando os desafios.
            </p>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
